test/parallel_api/iterator/iterators.pass.cpp

```diff
@@ -32,7 +32,7 @@
 using namespace TestUtils;
 
 //common checks of a random access iterator functionality
-template <typename RandomIt>
+template <bool check_default_constructible = true, typename RandomIt>
```

Comment:
This bool variable looks a little bit hacky to me. Can we do something better?

Reply:
Perhaps you have a suggestion?  

Basically, we are conveying to the random iterator check that we are willing to skip checking the default constructible check for specific cases like if the lambda is not default constructible.  To me, this seems like what we want to do, maybe it would be slightly better to have `skip_default_constructible_check` and default to `false`.

Otherwise, I suppose we could separate the default constructible check out and replicate it for each iterator?  I'd prefer this `bool` to replicated code or creating another function.  As mentioned in my other comment, `LegacyRandomAccessIterator` does include default constructible, so it seems to belong in this check.

Reply:
> Perhaps you have a suggestion?

Nothing is particular. Perhaps just making those checks separately... But I have not though deeply about that. I kind of would like to see simpler code. It took me some time to understand what exactly you are trying to test for. If everybody but me feels that the simplification is not needed I am fine with that.

Reply:
I've separated out the check for default constructible and replaced it with a note on random access iterators in oneDPL.
  
I've confirmed that a default constructible check is present for the iterator types I think we care about being default constructible, and made some minor changes to `iterator.pass`.